Lib/ruby/rubywstrings.swg

```diff
@@ -1,71 +1,55 @@
 /* -----------------------------------------------------------------------------
  * rubywstrings.swg
  *
- * Currently, Ruby does not support Unicode or WChar properly, so these
- * are still treated as char arrays for now.
- * There are other libraries available that add support to this in
- * ruby including WString, FXString, etc.
- * ----------------------------------------------------------------------------- */
-
-/* ------------------------------------------------------------
  *  utility methods for wchar_t strings 
  * ------------------------------------------------------------ */
 
 %fragment("SWIG_AsWCharPtrAndSize","header",fragment="<wchar.h>",fragment="SWIG_pwchar_descriptor",fragment="SWIG_AsCharPtrAndSize") {
 SWIGINTERN int
 SWIG_AsWCharPtrAndSize(VALUE obj, wchar_t **cptr, size_t *psize, int *alloc)
 {
-  return SWIG_AsCharPtrAndSize( obj, (char**)cptr, psize, alloc);
-//   VALUE tmp = 0;
-//   bool ok = false;
-//   if ( TYPE(obj) == T_STRING ) {
-//     if (cptr) {
-//       obj = tmp = SWIG_Unicode_FromObject(obj);
-//       ok = true;
-//     }
-//   }
-//   if (ok) {
-//     Py_ssize_t len = PyUnicode_GetSize(obj);
-//     rb_notimplement();
-//     if (cptr) {
-//       *cptr = %new_array(len + 1, wchar_t);
-//       SWIG_Unicode_AsWideChar((PyUnicodeObject *)obj, *cptr, len);
-//       (*cptr)[len] = 0;
-//     }
-//     if (psize) *psize = (size_t) len + 1;
-//     if (alloc) *alloc = cptr ? SWIG_NEWOBJ : 0;
-//     return SWIG_OK;
-//   } else {
-//     swig_type_info* pwchar_descriptor = SWIG_pwchar_descriptor();
-//     if (pwchar_descriptor) {
-//       void * vptr = 0;
-//       if (SWIG_ConvertPtr(obj, &vptr, pwchar_descriptor, 0) == SWIG_OK) {
-// 	if (cptr) *cptr = (wchar_t *)vptr;
-// 	if (psize) *psize = vptr ? (wcslen((wchar_t *)vptr) + 1) : 0;
-// 	return SWIG_OK;
-//       }
-//     }
-//   }
-//   return SWIG_TypeError;
+
+  if (TYPE(obj) == T_STRING) {
+    VALUE tmp = rb_str_conv_enc(obj, rb_enc_get(obj),
+                                rb_to_encoding(rb_str_new_cstr( SWIG_RUBY_WSTRING_ENCODING )) );
+    wchar_t* cstr = (wchar_t*) StringValuePtr(tmp);
+    size_t   size = RSTRING_LEN(tmp) / sizeof(wchar_t) + 1;
+
+    if( RSTRING_LEN(tmp) % sizeof(wchar_t) != 0 ) {
```

Comment:
Your patch has a mixture of `if (` and `if(`.  Sadly the SWIG code base doesn't consistently use either, but it's particularly unhelpful to mix them within a single function like this.  It seems `if (` dominates, so I'd go for that consistently.